Mathlib/Topology/CWComplex.lean

```diff
@@ -0,0 +1,164 @@
+
+/-- A CW-complex is a relative CW-complex whose (-1)-skeleton is empty. -/
+abbrev CWComplex := RelativeCWComplex (TopCat.of Empty)
+
+namespace CWComplex
+
+noncomputable section Topology
+
+/-- The inclusion map from `X` to `X'`, given that `X'` is obtained from X by attaching
+(n+1)-disks -/
+def attachCellsInclusion (X X' : TopCat) (n : ℤ) (att : AttachCells X X' n) : X ⟶ X' :=
+  @Limits.pushout.inr TopCat _ _ _ X
+    (sigmaSphereInclusion n att.cells)
+    (sigmaAttachMap X n att.cells att.attach_maps) _ ≫ att.iso_pushout.inv
+
+/-- The inclusion map from the n-skeleton to the (n+1)-skeleton of a relative CW-complex -/
+def skeletaInclusion {A : TopCat} (X : RelativeCWComplex A) (n : ℤ) : X.sk n ⟶ X.sk (n + 1) :=
+  attachCellsInclusion (X.sk n) (X.sk (n + 1)) n (X.attach_cells n)
+
+/-- The inclusion map from the n-skeleton to the m-skeleton of a relative CW-complex -/
+def skeletaInclusion' {A : TopCat} (X : RelativeCWComplex A)
+    (n : ℤ) (m : ℤ) (n_le_m : n ≤ m) : X.sk n ⟶ X.sk m :=
+  if h : n = m then by
+    subst m
+    exact 𝟙 (X.sk n)
+  else by
+    have h' : n < m := Int.lt_iff_le_and_ne.mpr ⟨n_le_m, h⟩
+    exact skeletaInclusion X n ≫ skeletaInclusion' X (n + 1) m h'
+  termination_by Int.toNat (m - n)
+  decreasing_by
+    simp_wf
+    rw [Int.toNat_of_nonneg (Int.sub_nonneg_of_le h')]
+    linarith
+
+/-- The colimit diagram in the definition of a relative CW-complex -/
+def colimitDiagram {A : TopCat} (X : RelativeCWComplex A) : ℤ ⥤ TopCat where
+  obj := X.sk
+  map := @fun n m n_le_m => skeletaInclusion' X n m <| Quiver.Hom.le n_le_m
+  map_id := by simp [skeletaInclusion']
+  map_comp := by
+    let rec p (n m l : ℤ) (n_le_m : n ≤ m) (m_le_l : m ≤ l) (n_le_l : n ≤ l) :
+        skeletaInclusion' X n l n_le_l =
+        skeletaInclusion' X n m n_le_m ≫
+        skeletaInclusion' X m l m_le_l :=
+      if hnm : n = m then by
+        unfold skeletaInclusion'
+        subst hnm
+        simp only [eq_mpr_eq_cast, ↓reduceDite, cast_eq, Category.id_comp]
+      else by
+        have h1 : n < m := Int.lt_iff_le_and_ne.mpr ⟨n_le_m, hnm⟩
+        have h2 : n < l := by linarith
+        unfold skeletaInclusion'
+        simp [hnm, Int.ne_of_lt h2]
+        by_cases hml : m = l
+        · subst hml
+          simp only [↓reduceDite, Category.comp_id]
+        congr
+        rw [p (n + 1) m l h1 m_le_l h2]
+        congr
+        simp only [hml, ↓reduceDite]
+        conv => lhs; unfold skeletaInclusion'
+        simp only [hml, ↓reduceDite]
+      termination_by Int.toNat (l - n)
+      decreasing_by
+        simp_wf
+        rw [Int.toNat_of_nonneg (Int.sub_nonneg_of_le h2)]
+        linarith
+    intro n m l n_le_m m_le_l
+    have n_le_m := Quiver.Hom.le n_le_m
+    have m_le_l := Quiver.Hom.le m_le_l
+    exact p n m l n_le_m m_le_l (Int.le_trans n_le_m m_le_l)
+
+/-- The topology on a relative CW-complex -/
+def toTopCat {A : TopCat} (X : RelativeCWComplex A) : TopCat :=
```

Comment:
This should be in the `RelativeCWComplex` namespace.

Reply:
Thanks. This definition has now been renamed `RelativeCWComplex.toTopCat`.
https://github.com/leanprover-community/mathlib4/blob/3cfda01c25cc8ef045b74478e0ccc1f6dbb3571c/Mathlib/Topology/CWComplex.lean#L132-L134